### PR TITLE
UNR-2386: Move load balance settings to ASpatialWorldSettings

### PR DESCRIPTION
**Pull request by yumnuska**
#### Description
- Add ASpatialWorldSettings, derived from AWorldSettings, as a place to specify per-level settings.
- Move LoadBalanceStrategy and LockPolicy from GDKSettings to SpatialWorldSettings.
- Update NetDriver to read from SpatialWorldSettings for these configurations.

#### Release note
Added to changelog

#### Tests
See related PR in TestGyms project (https://github.com/spatialos/UnrealGDKTestGyms/pull/21). HandoverGym is configured using this.

#### Documentation
TBD

#### Primary reviewers
@jennifer-at-improbable-io

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -46,6 +46,7 @@
 #include "Settings/LevelEditorPlaySettings.h"
 #include "SpatialGDKServicesModule.h"
 #endif
+#include "SpatialWorldSettings.h"
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Could we add this below "#include "SpatialGDKSettings.h""?

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ Usage: `DeploymentLauncher createsim <project-name> <assembly-name> <target-depl
 - Added the `bAsyncLoadNewClassesOnEntityCheckout` setting to SpatialGDKSettings that allows loading new classes asynchronously when checking out entities. This is off by default.
 - Added `IndexYFromSchema` functions for the `Coordinates`, `WorkerRequirementSet`, `FRotator`, and `FVector` classes. Remapped the `GetYFromSchema` functions for the same classes to invoke `IndexYFromSchema` internally, in line with other implementations of the pattern.
 - The logic responsible for taking an Actor and generating the array of Components that represents it as an Entity in SpatialOS has been extracted into `EntityFactory`.
+- Load Balancing Strategies and Locking Strategies can be set per-level using SpatialWorldSettings.
```
Comment (MatthewSandfordImprobable) [inline CHANGELOG.md]: I wonder if there are any docs that need to be updated too as a result of this change?

Comment (MatthewSandfordImprobable) [inline CHANGELOG.md]: I wonder if there are any docs that need to be updated too as a result of this change?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Could we add this below "#include "SpatialGDKSettings.h""?

File: SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h [hunk 3]
```diff
@@ -218,12 +218,6 @@ class SPATIALGDK_API USpatialGDKSettings : public UObject
 	UPROPERTY(EditAnywhere, Config, Category = "Load Balancing", meta = (EditCondition = "bEnableUnrealLoadBalancer"))
```
[the remark this answers is not in the document]
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h]: Oh I guess it can't due to it's nature. Ahh well...

File: CHANGELOG.md (hunk 2 above)
Comment (yumnuska) [reply CHANGELOG.md]: There are, @alastairdglennie has volunteered to update the doc he's building. I'll coordinate that once this merges.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -509,6 +475,51 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	PackageMap->Init(this, &TimerManager);
 }
 
+void USpatialNetDriver::CreateAndInitializeLoadBalancingClasses()
+{
+	const USpatialGDKSettings* SpatialSettings = GetDefault<USpatialGDKSettings>();
+	if (SpatialSettings->bEnableUnrealLoadBalancer == false)
```
Comment (jennifer-at-improbable-io) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: minor nit - I'd move the if check to where CreateAndInitializeLoadBalancingClasses is called, and just check the flag here.

**Pull request merged**